tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionRewriteSuite.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class RegularExpressionRewriteSuite extends AnyFunSuite {
+
+  private def verifyRewritePattern(patterns: Seq[String], excepted: Seq[RegexOptimizationType]): 
+      Unit = {
+    val results = patterns.map { pattern =>
+      val ast = new RegexParser(pattern).parse()
+      RegexRewriteUtils.matchSimplePattern(ast)
+    }
+    assert(results == excepted)
+  }
+
+  test("regex rewrite startsWith") {
+    import RegexOptimizationType._
+    val patterns = Seq("^abc.*", "\\A(abc).*", "^(abc).*def", "\\Aabc.*(.*).*", "^abc.*(.*).*", 
+        "^(abc)\\Z)")
```

Comment:
nit: same in Scala, we can use raw literals
```suggestion
    val patterns = Seq("^abc.*", raw"\A(abc).*", "^(abc).*def", raw"\Aabc.*(.*).*", "^abc.*(.*).*", 
        raw"^(abc)\Z)")
```

Reply:
done.